test/success/semi-simplicial.prl

```diff
@@ -61,20 +61,16 @@ Thm Pick/compose : [
     [ lam c, p1, p2. elim p2
     , lam c. fresh c', c'/ih -> elim c;
       [ lam p1. elim p1
-      , lam p1. unfold Pick in p1; reduce in p1;
-        let < head = p1/h, tail = p1/t > = p1. elim p1/h; reduce in p1/t;
-        [ lam p2. unfold Pick in p2; reduce in p2;
-          let < head = p2/h, tail = p2/t > = p2. elim p2/h; reduce in p2/t;
-          unfold Pick in goal; reduce in goal;
+      , lam p1. let < head = p1/h, tail = p1/t > = p1. elim p1/h;
+        [ lam p2. let < head = p2/h, tail = p2/t > = p2. elim p2/h;
```

Comment:
Maybe it is even better to run the pattern matching without leaving the type of `p1` expanded?